### PR TITLE
DCAR: Update dark mode colour for Match Stats background and % icon

### LINKED ISSUE #11318: [Bug]: Unreadable match stats in dark colour scheme
### What happened?

The Match Stats background colour does not account for the dark colour scheme.

image

A dark blue for `--match-stats-background` & `fill: currentColor` on the svg percentage sign seem to make this readable again:

image

### Where is this happening?

https://www.theguardian.com/football/2024/apr/30/bayern-munich-real-madrid-champions-league-match-report?dcr=apps&edition=uk

### What is the impact?

Unreadable match stats when users have a dark colour scheme.

### PR DESCRIPTION
**Pull request by domlander**
## What does this change?

Adds a dark mode colour for Match Stats background and the % icon in the possession doughnut

Fixes #11318 

## Why?

This component currently looks terrible and has poor colour contrast in dark mode. 

Note that this is not yet going live to users: [3036](https://github.com/guardian/mobile-apps-api/pull/3036), but with this we are better prepared for when we do.

## How to test

[Example localhost page](http://localhost:3030/AppsArticle/https://www.theguardian.com/football/2024/apr/30/bayern-munich-real-madrid-champions-league-match-report). Switch to dark mode on your browser or OS.

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before1][] | ![after1][] |
| ![before2][] | ![after2][] |

[before1]: https://github.com/guardian/dotcom-rendering/assets/9574885/55d13331-5fed-4558-961a-1e9a20353b28
[after1]: https://github.com/guardian/dotcom-rendering/assets/9574885/4beeb826-131c-44af-bb49-dde8df54331e
[before2]: https://github.com/guardian/dotcom-rendering/assets/9574885/7c46aaa7-ac4d-4937-b559-264efa767b20
[after2]: https://github.com/guardian/dotcom-rendering/assets/9574885/f394ff97-3567-4744-942c-9e92fd56b2c1

### CODE REVIEW

File: dotcom-rendering/src/components/Doughnut.tsx [hunk 1]
```diff
@@ -27,9 +27,10 @@ const SEGMENT_GAP = 2;
 const TAU = Math.PI * 2;
 const QUARTER_TURN = TAU / 4;
 
-const unitStyles = css`
+const percentageStyles = css`
 	${headlineBold34}
 	text-anchor: middle;
+	fill: var(--doughnut-percentage-icon);
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Doughnut.tsx:33]: I think inheritance is superior, here!

```suggestion
	fill: currentColor;
```

File: dotcom-rendering/src/palette.ts [hunk 2]
```diff
@@ -3763,6 +3763,9 @@ const matchStatsBackground: PaletteFunction = ({ design }) => {
 	}
 };
 
+const matchStatsBackgroundDark: PaletteFunction = () =>
+	sourcePalette.sport[300];
```
Comment (mxdvl) [inline dotcom-rendering/src/palette.ts:3767]: Feels a bit bright to my eye 👁️

**Review by mxdvl — APPROVED** [review]
Readable!

I'd consider not creating a new colour variable for the text inside the SVG — the rest of the match stats inherits its colour from the top level already without an explicit flag. It cannot think of a case where setting an SVG `text` to `currentColor` isn't the best default.

File: dotcom-rendering/src/palette.ts (hunk 2 above)
Comment (domlander) [reply dotcom-rendering/src/palette.ts:3767]: This does look on the bright side for a dark mode. As these pages are not rendered by DCAR yet, let's revisit when we are ready for this to go live.

**Pull request merged**